android/envoy/src/main/kotlin/org/greatfire/envoy/EnvoyPrefs.kt

```diff
@@ -0,0 +1,106 @@
+
+package org.greatfire.envoy
+
+import android.content.SharedPreferences
+import android.util.Log
+import androidx.preference.PreferenceManager
+
+/*
+    Stuff Envoy needs to store long term. Currently this is just URL failure
+    info
+
+    should this be updated to use the DataStore?
+    https://developer.android.com/topic/libraries/architecture/datastore
+*/
```

Comment:
it sounds like data store might handle concurrency somewhat better, but im not sure the other advantages are applicable here

Reply:
Yeah, that's something we can look at in the future, I just wanted to abstract out the access to the prefs so that's easier in the future (and so it was out of the way of the other code)